infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py

```diff
@@ -9,86 +9,122 @@
         settings_bucket_name = f"s3-{project_name}-settings-{stage_name}"
         settings_bucket = s3.Bucket(
             self,
             "salmonSettingsBucket",
             bucket_name=settings_bucket_name,
-            block_public_access=s3.BlockPublicAccess.BLOCK_ALL
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
         )
 
         # S3 settings files deployment
         s3deploy.BucketDeployment(
-            self, "salmonSettingsDeployment",
-            sources=[s3deploy.Source.asset('../config/settings')],
+            self,
+            "salmonSettingsDeployment",
+            sources=[s3deploy.Source.asset("../config/settings")],
             destination_bucket=settings_bucket,
-            destination_key_prefix='settings'
+            destination_key_prefix="settings",
         )
 
         # EventBridge Bus
         notification_bus = events.EventBus(
             self,
             "salmonNotificationsBus",
-            event_bus_name=f"eventbus-{project_name}-notification-{stage_name}"
+            event_bus_name=f"eventbus-{project_name}-notification-{stage_name}",
+        )
+
+        # EventBridge bus resource policy
+        # TODO: reuse existing settings reader
+        general_settings_file_path = "../config/settings/general.json"
+        with open(general_settings_file_path) as f:
+            try:
+                general_config = json.load(f)
+            except json.decoder.JSONDecodeError as e:
+                raise json.decoder.JSONDecodeError(
+                    f"Error parsing JSON file {general_settings_file_path}",
+                    e.doc,
+                    e.pos,
+                )
+
+        monitored_account_ids = [
+            account["AccountId"] for account in general_config["monitored_environments"]
+        ]
+        monitored_principals = [
+            iam.AccountPrincipal(account_id) for account_id in monitored_account_ids
+        ]
+
+        notification_bus.add_to_resource_policy(
+            iam.PolicyStatement(
+                sid="AllowMonitoredAccountsPutEvents",
+                actions=["events:PutEvents"],
+                effect=iam.Effect.ALLOW,
+                resources=[notification_bus.event_bus_arn],
+                principals=monitored_principals,
+            )
         )
 
         # Notification Lambda Eventbridge rule
         notification_lambda_event_rule = events.Rule(
             self,
             "salmonNotificationLambdaEventRule",
-            rule_name=f"eventbusrule-{
-                project_name}-notification-lambda-{stage_name}",
+            rule_name=f"eventbusrule-{project_name}-notification-lambda-{stage_name}",
             event_bus=notification_bus,
-            event_pattern=events.EventPattern(source=events.Match.prefix("aws"))
+            event_pattern=events.EventPattern(source=events.Match.prefix("aws")),
         )
 
         # Notification Lambda Role
         notification_lambda_role = iam.Role(
```

Comment:
based on sync-up conversation - looks like we need to rename it to alerts-lambda as notification-lambda is a different entity which actually sends notifications